### ISSUE TITLE
Cache without compression

### ISSUE THREAD

**Issue by thisismydesign** [actions/cache#161]
The file I'd like to cache is already generated compressed and I'd like to avoid double compression.

Compression might just be there to make the folder structure to be cached easier to handle. In this case, I'd point to the request to be able to cache a single file: https://github.com/actions/cache/issues/33

**Comment by thisismydesign** [comment]
Still relevant

**Comment by bishal-pdMSFT** [comment]
> Compression might just be there to make the folder structure to be cached easier to handle
True but there are additional factors as well - compression reduced the size and that has lots of benefit - upload and download becomes faster, storage is optimally used (without compression the per repo quota may fill up faster), and if I am not wrong the checksum check is more efficient and robust.

We could argue about double compression but unless the double compression is having much performance impact, I would not trade it for simplicity and uniformity in how we handle files/folders.

**Comment by vsvipul** [comment]
+1 to what @bishal-pdMSFT  mentioned. Also Single file cache support was added in [v2](https://github.com/actions/cache/issues/323). Implementing this changes will require considerable effort from our end and the end result will only avoid double compression and some saving of minutes only if user is already providing a compressed file. We have made a note of this and can consider in the future if there is a lot of ask from the community. Closing this for now. Thank you for bringing this up.
